futures-util/src/stream/stream/peek.rs

```diff
@@ -120,9 +125,51 @@ impl<S: Stream> Stream for Peekable<S> {
 // Forwarding impl of Sink from the underlying stream
 #[cfg(feature = "sink")]
 impl<S, Item> Sink<Item> for Peekable<S>
-    where S: Sink<Item> + Stream
+where
+    S: Sink<Item> + Stream,
 {
     type Error = S::Error;
 
     delegate_sink!(stream, Item);
 }
+
+/// Future for the [`peek()`] function from [`Peekable`]
+#[must_use = "futures do nothing unless polled"]
+pub struct Peek<'a, St: Stream> {
+    inner: Option<Pin<&'a mut Peekable<St>>>,
+}
+
+impl<St: Stream> Unpin for Peek<'_, St> {}
+
+impl<St: Stream> fmt::Debug for Peek<'_, St> {
```

Comment:
If you add `where St: fmt::Debug + St::Item: fmt::Debug` then you can add the `inner` field to the debug output.